test_programs/execution_success/reference_counts/src/main.nr

```diff
@@ -76,3 +82,30 @@ fn assert_refcount<T>(array: [T; 3], expected: u32) {
         assert_eq(count, 0);
     }
 }
+
+unconstrained fn regression_7297() {
+    println("regression_7297");
+    let mut array = [0, 1, 2];
+
+    let refcount_0 = array_refcount(array);
+    borrow_mut_two(&mut array, &mut array, refcount_0);
+    let refcount_1 = array_refcount(array);
+    let array_2 = copy_mut(array, refcount_1);
+    let refcount_2 = array_refcount(array);
+
+    println([refcount_0, refcount_1, refcount_2]);
+
+    // Mutation of the original could occur if we double decremented the RC and then went back to 1 by accident.
+    // For this to come out we have to run the test with `--inliner-aggressiveness -9223372036854775808`
```

Comment:
Does this fail only with `--enable-brillig-debug-assertions` or without it as well? According to https://github.com/noir-lang/noir/pull/7297#issuecomment-2643007393 it sounds like you were able to trigger a failure without the extra assertions?

Reply:
Yes it does fail without `--enable-brillig-debug-assertions` if `--inliner-aggressiveness` is at the minimum.

These tests are here to show that we can observe illegal behaviour, such as modifying the original array instead of a copy. The `--enable-brillig-debug-assertions` changes the failure so that instead of "expected 6 but got 4" we have "ref-count went to zero". The tests were added to show it's actually indicating the presence of a correctness bug that exists. 

Otherwise we could argue that ref-count=0 could be considered going out of scope, but that's not the case in our code at the moment; plus I verified that if I put the 0 assertion _before_ the decrement, which is another way to detect would-be underflows, we still fail all tests in aztec-packages.